File: app/src/main/java/com/ericho/coupleshare/ServerUrlAdapter.java
package com.ericho.coupleshare;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.ericho.coupleshare.adapter.BaseObjectAdapter;
import com.ericho.coupleshare.setting.model.ServerBean;
import com.ericho.coupleshare.util.DrawableUtil;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by EricH on 18/4/2017.
 */

public class ServerUrlAdapter extends BaseObjectAdapter<ServerBean> {
    public ServerUrlAdapter(Context context, List<ServerBean> items) {
        super(context, items);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ServerBean i = getItem(position);
        ViewHolder holder;
        if(convertView == null){
            convertView = layoutInflater.inflate(R.layout.grid_change_server,parent,false);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        }else {
            holder = (ViewHolder) convertView.getTag();
        }
        Drawable drawable = DrawableUtil.getDrawble(getContext(),i.getResourceInteger());
        holder.imageView.setImageDrawable(drawable);
        holder.text1.setText(i.getDisplayName());


        return convertView;
    }

    static class ViewHolder{
        @BindView(R.id.image) protected ImageView imageView;
        @BindView(R.id.text1) protected TextView text1;
        ViewHolder(View v){
            ButterKnife.bind(this,v);
        }
    }
}

File: app/src/main/java/com/ericho/coupleshare/adapter/ServerUrlAdapter.kt
package com.ericho.coupleshare.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.ericho.coupleshare.R
import com.ericho.coupleshare.setting.model.ServerBean
import com.ericho.coupleshare.util.DrawableUtil

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.adapter
 */
class ServerUrlAdapter constructor(context: Context ,items:List<ServerBean>?):BaseObjectAdapter<ServerBean> (context,items) {


    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val i = getItem(position)
        val holder: ViewHolder
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.grid_change_server, parent, false)
            holder = ViewHolder(convertView)
            convertView!!.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }
        val drawable = DrawableUtil.getDrawble(getContext(), i.resourceInteger)
        holder.imageView!!.setImageDrawable(drawable)
        holder.text1!!.text = i.displayName


        return convertView
    }

    class ViewHolder(v: View) {
        var imageView: ImageView = v.findViewById<ImageView>(R.id.image) as ImageView
        var text1: TextView = v.findViewById<ImageView>(R.id.text1) as TextView

    }
}